text/0036-async-testbench-functions.md

```diff
@@ -147,62 +149,86 @@ The following `Simulator` methods have their signatures updated:
+- `process_type`
+  - Property, `ProcessType`, indicates the type of the current process.
+    This property allows a generic simulation helper function to assert that it's being run under the appropriate process type.
 - `get(expr: Value) -> int`
 - `get(expr: ValueCastable) -> any`
   - Returns the value of `expr` when awaited.
-    When `expr` is a value-castable, the value will be converted through `.from_bits()`.
+    When `expr` is a value-castable, and its `shape()` is a `ShapeCastable`, the value will be converted through the shape's `.from_bits()`.
+    Otherwise, a plain integer is returned.
 - `set(expr: Value, value: ConstLike)`
 - `set(expr: ValueCastable, value: any)`
   - Set `expr` to `value` when awaited.
-    When `expr` is a value-castable, the value will be converted through `.const()`.
-- `memory_read(instance: MemoryInstance, address: int)`
+    When `expr` is a value-castable, and its `shape()` is a `ShapeCastable`, the value will be converted through the shape's `.const()`.
+    Otherwise, it must be a const-castable `ValueLike`.
+- `memory_read(instance: MemoryIdentity, address: int)`
   - Read the value from `address` in `instance` when awaited.
-- `memory_write(instance: MemoryInstance, address: int, value: int, mask:int = None)`
+- `memory_write(instance: MemoryIdentity, address: int, value: int, mask:int = None)`
   - Write `value` to `address` in `instance` when awaited. If `mask` is given, only the corresponding bits are written.
-- `delay(interval: float)`
-  - Create a trigger object for advancing simulation by `interval` seconds.
+    Like `MemoryInstance`, these two functions are an internal interface that will be usually only used via `lib.Memory`.
+    It comes without a stability guarantee.
 - `tick(domain="sync", *, context=None)`
-  - Create a trigger object for advancing simulation until the next active edge of the `domain` clock.
+  - Create a domain trigger object for advancing simulation until the next active edge of the `domain` clock.
     When an elaboratable is passed to `context`, `domain` will be resolved from its perspective.
   - If `domain` is asynchronously reset while this is being awaited, `amaranth.sim.AsyncReset` is raised.
+- `delay(interval: float)`
+  - Create a combinable trigger object for advancing simulation by `interval` seconds.
 - `changed(*signals)`
-  - Create a trigger object for advancing simulation until any signal in `signals` changes.
+  - Create a combinable trigger object for advancing simulation until any signal in `signals` changes.
 - `edge(signal, value)`
-  - Create a trigger object for advancing simulation until `signal` is changed to `value`.
+  - Create a combinable trigger object for advancing simulation until `signal` is changed to `value`.
     `signal` must be a 1-bit signal or a 1-bit slice of a signal.
+    `value` is a boolean where true indicates rising edge and false indicates falling edge.
```

Comment:
Is an `int` also acceptable? The code below suggests that yes, this statement is conflicting with it. (So I guess the type annotation would be `value: int` and also `value in (0, 1)` is required.)